src/video_core/shader/decode/arithmetic_integer.cpp

```diff
@@ -138,6 +138,32 @@ u32 ShaderIR::DecodeArithmeticInteger(NodeBlock& bb, u32 pc) {
         SetRegister(bb, instr.gpr0, value);
         break;
     }
+    case OpCode::Id::ICMP_CR:
+    case OpCode::Id::ICMP_R:
+    case OpCode::Id::ICMP_RC: {
+        UNIMPLEMENTED_IF(instr.icmp.is_signed != 0);
+        const Node zero = Immediate(0);
+
+        const auto [op_a, op_b] = [&]() -> std::tuple<Node, Node> {
```

Comment:
```suggestion
        const auto [op_a, op_b] = [&]() -> std::pair<Node, Node> {
```

generally requires less template instantiation work by the compiler.